python/sdss_access/path/path.py

```diff
@@ -1529,34 +1557,200 @@ def pad_fieldid(self, filetype, **kwargs):
-        if not tileid:
-            return '0000XX'
-        elif '*' in str(tileid):
-            return '{0}XX'.format(tileid)
-        return '{:0>4d}XX'.format(int(tileid) // 1000)
+        run2d = kwargs.get('run2d', None)
+        
+        if (not run2d):
+            return ''
+        elif ('v5' in run2d) or (str(run2d) in ['26','103','104']):
+            return ''
+        elif ('v6_0' in run2d) or ('v6_1' in run2d):
+            if filetype.lower() in ['speclite_epoch','specfull_epoch',
+                                    'spallfield_epoch','spalllinefield_epoch']:
+                return('epoch/spectra')
+            else:
+                return ''
+        else:
+            if filetype.lower() in ['fieldlist_epoch','spall_epoch',
+                                    'spall-lite_epoch','spallline_epoch']:
+                return('summary/epoch')
+            elif filetype.lower() in ['speclite_epoch','specfull_epoch',
+                                      'spallfield_epoch','spalllinefield_epoch']:
+                return('spectra/epoch')
+            elif filetype.lower() in ['conflist','fieldlist','spall',
+                                      'spall-lite','spallline']:
+                return('summary/daily')
+            elif filetype.lower() in ['speclite','specfull',
+                                     'spallfield','spalllinefield']:
+                return('daily')
+            else:
+                return('fields')
+                    
+    def spcoaddobs(self, filetype, **kwargs):
+        ''' Returns the formatted observatory flag for custom coadds for the BOSS idlspec2d
+        
+        Parameters
+        ----------
+        filetype : str
+            File type parameter
+        run2d : str
+            BOSS idlspec2d run2d version
+        obs : str
+            Observatory of observations: LCO, APO, ''(for merged)
+            
+        Returns
+        -------
+        obs : str
+        '''
+        
+        obs = kwargs.get('obs', None)
+        run2d = kwargs.get('run2d', None)
+        
+        if not obs:
+            return ''
+            
+        elif obs == '':
+            return ''
+        elif (('v6_0' in run2d) or ('v6_1' in run2d)) and obs.lower() == 'apo':
+            return ''
+        elif obs == '*':
+            return obs
+        else:
+            return '_{}'.format(obs.lower())
+            
+    def epochflag(self, filetype, **kwargs):
+        ''' Returns the flag for epoch coadds for the BOSS idlspec2d
+        
+        Parameters
+        ----------
+        filetype : str
+            File type parameter
+        run2d : str
+            BOSS idlspec2d run2d version
+            
+        Returns
+        -------
+        epochflag : str
+        '''
+        
+        run2d = kwargs.get('run2d', None)
+        
+        if (('v6_0' in run2d) or ('v6_1' in run2d)):
+            return ''
+        else:
+            return '-epoch'
```

Comment:
Don't need the else here.  
```suggestion
        return '-epoch'
```

Reply:
fixed